src/main/java/bio/terra/janitor/service/cleanup/FlightManager.java

```diff
@@ -197,32 +197,52 @@ public int updateCompletedFlights(int limit) {
    */
   private boolean completeFlight(TrackedResourceAndFlight resourceAndFlight) {
     String flightId = resourceAndFlight.cleanupFlight().flightId();
-    FlightState flightState;
+    Optional<FlightStatus> flightStatus;
     try {
-      flightState = stairway.getFlightState(flightId);
+      flightStatus = Optional.of(stairway.getFlightState(flightId).getFlightStatus());
+    } catch (FlightNotFoundException e) {
```

Comment:
I don't quite understand when and where FlightNotFoundException is throwed. Does line 202 throw this?

Reply:
Yes. Stairway#getFlightState can throw an unchecked FlightNotFoundException if it doesn't have a record of the flightId. That makes sense given the type signature (what else should it return if it doesn't find a flight? arguably null), but I don't think it's great that nothing about that is documented in the function's type signature.